addon/templates/components/frost-modal-dialog.hbs

```diff
@@ -59,7 +59,16 @@
   {{! END-SNIPPET }}
 
   {{#if params.footer}}
-    <div class="frost-modal-dialog-footer-text">{{params.footer}}</div>
+    {{#if (type-of params.footer 'object')}}
+      <div class="frost-modal-dialog-footer-object"
+        data-test={{hook (concat hook '-object')}}>
+        {{component params.footer}}
```

Comment:
I'm confused if you use the component helper in the consuming case why do you need it here? You should just be able to do `{{params.footer}}`. I'm already successfully passing a component in to the footer without your changes, the only issue is prop-types complains.